source/include/platform/acmacosx.h

```diff
@@ -119,6 +119,7 @@
 #include "aclinux.h"
 
 #ifdef __APPLE__
+#include <semaphore.h>
 #define sem_destroy         sem_close
```

Comment:
That really sounds hackiy. :-)

Then IMO,
The first choice might be:
Converting sem_destroy to sem_close in osunixxf.c.

Does APPLE build always include <semaphore.h>?
I think the problem is the following line:
#include "aclinux.h"
If so, probably the second choice could be writing the following in acmacosx.h:
#define ACPI_USE_SYSTEM_CLIBRARY
#define ACPI_USE_STANDARD_HEADERRS

#ifdef ACPI_USE_STANDARD_HEADERRS
#include <semaphore.h>
#endif 

#include "aclinux.h"